server/src/admin-cli/commands.ts

```diff
@@ -0,0 +1,163 @@
+    console.error(e.message);
+  }
+}
+
+export async function upgradeUnvettedBeneficiaryCmd(app: App) {
+  try {
+    const { upgradeBy } = await prompt(prompts.upgradeUnvettedBeneficiary);
+    let user;
+    switch(upgradeBy) {
+      case SPECIFY_BENEFICIARY_BY_ID: 
+        const { id } = await prompt(prompts.specifyVettedBeneficiaryID);
+        user = await app.users.getById(id);
+        if (user) {
+          console.log(user);
+          await askForConfirmation('upgrade', 'id', null, user, app);
+        }
+        break;
+      case SPECIFY_BENEFICIARY_BY_PHONE:
+        const { phone } = await prompt(prompts.specifyVettedBeneficiaryPhone); 
+        user = await app.users.getByPhone(phone);
+        if (user) {
+          console.log(user);
+          await askForConfirmation('upgrade', 'phone', null, user, app);
+        }
+        break;
+    }
+  }
+  catch(e) {
+    console.error(e.message);
+  }
+}
+
+async function askForConfirmation(command: string, property: string, args: any = null, user: User, app: App) {
+  if (command === 'add') {
+    console.log(args);
+    prompts.confirmCommand[0].message = `Are you sure you want to add this beneficiary?`;
+    const { confirmation } = await prompt(prompts.confirmCommand);
+    if (confirmation) {
+      const user = await app.users.addUnvettedBeneficiary(args);
+      console.log(user);
+    }
+    else {
+      console.log('Command aborted!');
+    }
+  }
+  else if (command === 'upgrade' && property === 'id') {
+    prompts.confirmCommand[0].message = `Are you sure you want to upgrade ${user.name}?`;
+    const { confirmation } = await prompt(prompts.confirmCommand);
+    if (confirmation) {
+      await upgradeUnvettedBeneficiaryByProperty('id', user._id, app);
+    }
+    else {
+      console.log('Command aborted!');
+    }
+  }
+  else if (command === 'upgrade' && property === 'phone') {
+    prompts.confirmCommand[0].message = `Are you sure you want to upgrade ${user.name}?`;
+    const { confirmation } = await prompt(prompts.confirmCommand);
+    if (confirmation) {
+      await upgradeUnvettedBeneficiaryByProperty('phone', user.phone, app);
+    }
+    else {
+      console.log('Command aborted!');
+    }
+  }
+  else if (command === 'verify' && property === 'id') {
+    prompts.confirmCommand[0].message = `Are you sure you want to verify ${user.name}?`;
+    const { confirmation } = await prompt(prompts.confirmCommand);
+    if (confirmation) {
+      await verifyVettedBeneficiaryByProperty('id', user._id, app);
+    }
+    else {
+      console.log('Command aborted!');
+    }
+  }
+  else if (command === 'verify' && property === 'phone') {
+    prompts.confirmCommand[0].message = `Are you sure you want to verify ${user.name}?`;
+    const { confirmation } = await prompt(prompts.confirmCommand);
+    if (confirmation) {
+      await verifyVettedBeneficiaryByProperty('phone', user.phone, app);
+    }
+    else {
+      console.log('Command aborted!');
+    }
+  }
+}
```

Comment:
I think this would be better to understand and maintain if each command had its own `askForConfirmation` function as opposed to all those if-statements.

It also seems that there is a lot of duplicate code commands that accept either phone or id as property. I think you could combine the common part and only place the condition in the final function, e.g.:

```js
function askForConfirmationForVerifyCommand(property, args, user, app) {
    prompts.confirmCommand[0].message = `Are you sure you want to verify ${user.name}?`;
    const { confirmation } = await prompt(prompts.confirmCommand);
    if (confirmation) {
      const value = property === 'phone' ? user.phone : user._id;
      await verifyVettedBeneficiaryByProperty(property, value, app);
    }
    else {
      console.log('Command aborted!');
    }
}
```